src/Model/CompositeTypeGo.cs

```diff
@@ -75,15 +75,21 @@ public CompositeTypeGo(MethodGo responseToWrap)
                 throw new ArgumentException("{0} is not a valid type for SyntheticType", wrappedType.ToString());
             }
 
-            // gosdk: Ensure the generated name does not collide with existing type names
+            CodeModel = responseToWrap.CodeModel;
             BaseType = wrappedType;
 
             if (wrappedType.XmlIsWrapped)
             {
-                Name = wrappedType.XmlName;
+                // ensure that the name won't collide with an existing type.
+                // if it does then we'll give it the standard response name.
+                if (!CodeModel.ModelTypes.Where(mt => (mt.Name == wrappedType.XmlName)).Any())
+                {
+                    Name = wrappedType.XmlName;
+                }
                 XmlProperties = wrappedType.XmlProperties;
             }
-            else
+            
+            if (Name == null)
             {

```

Comment:
can this null Name can come only from the XmlName there ? If so maybe we can do a `wrappedType.XmlName ?? ` $"{responseToWrap.Name}Response"` there

Reply:
The null happens if we find an existing type that has the same name as ```wrappedType.XmlName``` (see line 85).